### DISCUSSION TITLE
Question: why is the coordinate system compiler time specific?

### DISCUSSION THREAD

**Discussion by kurtsansom** [PrincetonUniversity/athena#398] [Q&A] [answered]
What is the logic behind making the coordinate system compiler time specific?

Whats the limitation to having the coordinate system be chosen at run time?

**Comment by kurtsansom** [comment]
looking at the source code `configure.py` sets `COORDINATES_FILE` in the makefile.in but it doesn't appear to be used in the makefile ? 

So far I am seeing that it's `#define` usage, which makes me think it is possible to make a run time option.

**Comment by c-white** [comment, marked as answer]
This was done for optimization. The code only knows about the chosen coordinates through comparisons to `COORDINATE_SYSTEM`, especially in `meshblock.cpp`, and these could be set at runtime. But if the comparisons are not compile-time evaluatable, there is little chance the compiler will inline calls to functions in the Coordinates class.

If it turns out that with present-day compilers we can get the same performance by specifying coordinates at runtime, then we could consider changing this.
